packages/synapse-react-client/src/components/TimelinePlot/TimelinePlot.tsx

```diff
@@ -0,0 +1,243 @@
+  searchParams?: Record<string, string>
+  sqlOperator?: SQLOperator
+  defaultSpecies?: string //for test
+}
+export const TimelinePlot = ({
+  sql,
+  searchParams,
+  sqlOperator,
+  defaultSpecies,
+}: TimelinePlotProps) => {
+  // Fetch the table data
+  const eventsTableId = parseEntityIdFromSqlStatement(sql)
+  const [species, setSpecies] = useState<string | undefined | null>(
+    defaultSpecies,
+  )
+  const queryFilters =
+    getAdditionalFilters(eventsTableId, searchParams, sqlOperator) ?? []
+  const speciesFilter: ColumnSingleValueQueryFilter | undefined = species
+    ? {
+        columnName: 'species',
+        concreteType:
+          'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+        values: [species],
+        operator: ColumnSingleValueFilterOperator.EQUAL,
+      }
+    : undefined
+  const additionalFilters = [...queryFilters]
+  if (speciesFilter) {
+    additionalFilters.push(speciesFilter)
+  }
+  const eventTableQuery = useGetFullTableQueryResults(
+    {
+      entityId: eventsTableId,
+      query: {
+        sql: `${sql} WHERE observationTime IS NOT NULL`,
+        sort: [
+          {
+            column: 'observationTime',
+            direction: 'ASC',
+          },
+        ],
+        additionalFilters: additionalFilters,
+      },
+
+      partMask: BUNDLE_MASK_QUERY_RESULTS,
+      concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    },
+    {
+      enabled: !!species,
+    },
+  )
+
+  const { data: eventsData, isLoading } = eventTableQuery
+
+  if (isLoading) {
+    return <LoadingTimelinePlot />
+  }
+  const observationPhaseIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header => header.name.toLowerCase() === OBSERVATION_PHASE_COLUMN_NAME,
+    )!
+  const observationTimeIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header => header.name.toLowerCase() === OBSERVATION_TIME_COLUMN_NAME,
+    )!
+  const observationTimeUnitIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header =>
+        header.name.toLowerCase() === OBSERVATION_TIME_UNITS_COLUMN_NAME,
+    )!
+  const observationSubmitterNameIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header =>
+        header.name.toLowerCase() === OBSERVATION_SUBMITTER_NAME_COLUMN_NAME,
+    )!
+  const observationTextIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header => header.name.toLowerCase() === OBSERVATION_TEXT_COLUMN_NAME,
+    )!
+  const observationTypeIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header => header.name.toLowerCase() === OBSERVATION_TYPE_COLUMN_NAME,
+    )!
+  const submitterUserIdIndex =
+    eventsData?.queryResult?.queryResults.headers.findIndex(
+      header =>
+        header.name.toLowerCase() === OBSERVATION_SUBMITTER_USER_ID_COLUMN_NAME,
+    )!
+
+  const schema: ObservationCardSchema = {
```

Comment:
For challenges, some of this will need to change.  We'll probably need to support a generic card rather than this special ObservationCard that is being used by NF.